### ISSUE TITLE
Unexpected numbered list behavior

### ISSUE THREAD

**Issue by mjr6140** [silverbulletmd/silverbullet#465]
I added some nested numbered list to a page and the behavior didn't match my expectations for Markdown editors.

The first issue is that when nesting numbered lists the numbering continues as if it was a single list, vs. resetting back to "1" (or another identifier like "a").  In this example I would have expected "3" and "4" to either reset to "1", "2" again or to swap to something like "a", "b" (which would be my personal preference):

![image](https://github.com/silverbulletmd/silverbullet/assets/8962919/df65e53e-b522-4a16-ba31-a2d7a54a99fa)

The second issue was that after deleting an item in the middle of the list it did not automatically renumber subsequent items as I was expecting.  Same list from above but I deleted item 3 in the middle:

![image](https://github.com/silverbulletmd/silverbullet/assets/8962919/06a6066e-c336-4248-9dc9-70874c2698a2)

Wondering if this is something that we may be able to change?

**Comment by zefhemel** [comment]
Yeah, we may be able to change this, but for now we inherit all this behavior from @codemirror/lang-markdown: https://github.com/codemirror/lang-markdown.

At some point we may want to fork this behavior and change it, or reimplement custom for SB.
